src/DataStructures/VectorImpl.hpp

```diff
@@ -552,3 +552,47 @@ std::ostream& operator<<(std::ostream& os,
     }                                                    \
   };                                                     \
   }  // namespace MakeWithValueImpls
+
+// {@
+/*!
+ * \ingroup DataStructuresGroup
+ * \ingroup TypeTraitsGroup
+ * \brief Helper struct to determine the base storage type of a VectorImpl or
```

Comment:
What is the purpose of this struct?  I find the name completely unintuitive for what it does to a type.  The documentation talks about base storage type (what does that mean?) of a VectorImpl.
So why does it work on a non-VectorImpl at all?
It looks like it give `DataVector` for a `std::complex<DataVector>`

Reply:
The idea is just to have a type trait which gives the `ElementType` for a type which inherits from `VectorImpl`, or just get the type if it is a valid storage type, getting behavior analogous to `tt::get_fundamental_type`. The examples a few lines lower are hopefully slightly helpful.

Perhaps it would be better named `get_vector_element_type` and omit the inexact phrasing of "base" everywhere it appears?

At current, I have only used it in the generic vector math test utliity, so it could be moved to the appropriate detail namespace if it has inappropriate behavior for a public interface.

I agree the corner case of complex types with non-fundamental `value_type`s should be prohibited.